### ISSUE TITLE
v2022.6.1; Queue errors when adding single tracks 

### ISSUE THREAD

**Issue by erkr** [music-assistant/hass-music-assistant#392]
I was adding single tracks to the queue. Not sure if it is relevant, but I was doing that from a playlist, not in the tracks menu
That attempt failed, queue was corrupt (when opening animation kept floating).
These errors in the log:
```
Error doing job: Task exception was never retrieved
1[547128774608] Error handling message: Unknown error (unknown_error)
19:56:20 – (ERROR) Home Assistant WebSocket API (custom integration) - message first occurred at 19:56:08 and shows up 3 times
Error doing job: Task exception was never retrieved
19:56:08 – (ERROR) /usr/src/homeassistant/homeassistant/runner.py
```

details:
```
Logger: homeassistant.components.websocket_api.http.connection
Source: custom_components/mass/websockets.py:873
Integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 19:56:08 (3 occurrences)
Last logged: 19:56:20

[547128774608] Error handling message: Unknown error (unknown_error)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 27, in _handle_async_response
    await func(hass, connection, msg)
  File "/config/custom_components/mass/websockets.py", line 101, in async_get_mass_func
    await orig_func(hass, connection, msg, mass)
  File "/config/custom_components/mass/websockets.py", line 873, in websocket_playerqueue_items
    result = [x.to_dict() for x in queue.items]
  File "/config/custom_components/mass/websockets.py", line 873, in <listcomp>
    result = [x.to_dict() for x in queue.items]
AttributeError: 'NoneType' object has no attribute 'to_dict'
```
Logger: homeassistant
Source: /usr/src/homeassistant/homeassistant/runner.py:96
First occurred: 19:56:08 (1 occurrences)
Last logged: 19:56:08

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/music_assistant/models/player_queue.py", line 796, in _save_items
    "items": [x.to_dict() for x in self._items],
  File "/usr/local/lib/python3.9/site-packages/music_assistant/models/player_queue.py", line 796, in <listcomp>
    "items": [x.to_dict() for x in self._items],
AttributeError: 'NoneType' object has no attribute 'to_dict'
```

```

**Comment by erkr** [comment]
Related to TTS, or even the root casue of TTS issues; Errors are very similar

**Comment by erkr** [comment]
Test update with v2022.6.2;
The errors in the log are solved. 
Functionally it still doesn't work as expected. Only the last track manually added to the queue is there.
So I start with an empty queue, add an track (is added). Add another track, that overwrites the previous one!

**Comment by marcelveldt** [comment]
OK, this was a pain to reproduce. Turns out it only happens when shuffle is enabled....
I managed to fix it. The fix will be in 2022.6.5. Please reopen if there still is an issue.

**Comment by erkr** [comment]
You are a hero🙏
